src/shortcodes/instagram.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const {extractLayoutString} = require('../helpers/shortcodeLayoutParser');
+
+module.exports = (...args) => {
+  if (args.length == 0) {
+    throw new Error('Missing instagram id');
+  }
+  const id = args[0];
+  const layout = extractLayoutString(args.splice(1), {
+    layout: 'responsive',
+    width: 1,
```

Comment:
why 1? Would 0 get it ignored/stripped by amp?

Reply:
I assume default aspect ratio to be 1:1 (responsive layout takes care of sizing). Added a comment.